### PR TITLE
WBS 4.2.1/User management grid and search 

### PR DESCRIPTION
**Pull request by getnithesh**
User management grid react component, action and reducers.
Column filters , wild card search and filter.

### CODE REVIEW

File: src/components/UserManagement/UserTableFooter.jsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    pageCounter += 1;
+  }
+  return pageLinks;
+});
+
+/**
+ * Single page lin item 
+ */
+const PageLinkItem = React.memo((props) => {
+  return <a href="javascript:void(0)"
```
[the remark this answers is not in the document]
Comment (getnithesh) [reply src/components/UserManagement/UserTableFooter.jsx:111]: Resolved, handed in the event handler

File: src/reducers/allUserProfilesReducer.js [hunk 2]
```diff
@@ -1,7 +1,37 @@
-export const allUserProfilesReducer = (userProfiles = null, action) => {
-  if (action.type === 'GET_ALL_USER_PROFILES') {
```
[the remark this answers is not in the document]
Comment (getnithesh) [reply src/reducers/allUserProfilesReducer.js:2]: I had a scan in he code and this is not using anywhere, I shall remove it from the index.js

File: src/components/UserManagement/DropDownSearchBox.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      selectedValue: ''
+    }
+  }
+
+  render() {
+    return <select onChange={this.onSelectionChange}>
+      <option value={''} >{''}</option>
+      {this.props.items.map(item => {
+        return <option value={item} >{item}</option>
+      })}
```
[the remark this answers is not in the document]
Comment (getnithesh) [reply src/components/UserManagement/DropDownSearchBox.jsx:16]: Added the key.

File: src/components/UserManagement/UserTableFooter.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndonIgnatov) [reply src/components/UserManagement/UserTableFooter.jsx]: Looking good, Thx

File: src/components/UserManagement/DropDownSearchBox.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndonIgnatov) [reply src/components/UserManagement/DropDownSearchBox.jsx]: That's great! Thx

File: src/reducers/allUserProfilesReducer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndonIgnatov) [reply src/reducers/allUserProfilesReducer.js:2]: Sounds good!

**Pull request merged**